### PR TITLE
Overhaul Emacs Tutorial

### PR DESCRIPTION
**Pull request by awdeorio**
Overhaul the Emacs tutorial to be in sync with recent updates to our IDE tutorials.

### CODE REVIEW

**Comment by awdeorio** [comment]
@melodell Do you care to take a look at this?

File: docs/setup_emacs.md [hunk 1]
```diff
@@ -162,219 +156,258 @@ Learn a few more keyboard shortcuts with this [Emacs Cheat Sheet for beginners](
...
+### Caps lock as control
 You'll use the control key a lot in Emacs.  Some users find it more ergonomic to map the Caps Lock key to Control.  Here's the setting in MacOS.
 
 <img src="images/emacs022.png" width="512px" />
 
 
-## macOS
-On Apple laptops, it's more ergonomic to map Command to Meta and Option to Super.  Add this to your `init.el`.
+### macOS
+On Apple laptops, it's more ergonomic to map Command to Meta and Option to Super.  Add this to your `~/emacs.d/init.el`.
```
Comment (melodell) [inline docs/setup_emacs.md:172]:
```suggestion
On Apple laptops, it's more ergonomic to map Command to Meta and Option to Super.  Add this to your `~/.emacs.d/init.el`.
```

File: docs/setup_emacs.md [hunk 2]
```diff
@@ -162,219 +156,258 @@ Learn a few more keyboard shortcuts with this [Emacs Cheat Sheet for beginners](
...
 
-# Create a project
-Emacs doesn't require any special setup for a project.
-
-Create a new folder for your project. Your folder location might be different.
+## Customize
+The out-of-the-box Emacs configuration is terrible.  Some users start with *Spacemacs* or *Doom Emacs*, which are just Emacs shipped with a bunch of customizations.  Many of these users later graduate to vanilla Emacs with their own customizations.  Here's [awdeorio's init.el](https://github.com/awdeorio/dotfiles/blob/main/.emacs.d/init.el).
 
+### Basic
+Edit `~/emacs.d/init.el` with Emacs.  Yours will probably be blank.
```
Comment (melodell) [inline docs/setup_emacs.md:184]:
```suggestion
Edit `~/.emacs.d/init.el` with Emacs.  Yours will probably be blank.
```

**Review by melodell — COMMENTED** [review]
Left a few very minor suggestions! :)

I'm also wondering if moving the second pro-tip about the menu bars from [here](https://preview.sesh.rs/previews/eecs280staff/tutorials/120/setup_emacs.html#key-bindings) to somewhere down in the Customize section would flow more logically, since that's where editing `~/.emacs.d/init.el` officially gets introduced. (Feel free to ignore this though)

LGTM!

File: docs/setup_emacs.md [hunk 3]
```diff
@@ -527,26 +543,14 @@ For a light weight text-only Emacs install on a Linux server, use the `emacs-nox
 ```console
 $ sudo apt-get install emacs-nox
```
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_emacs.md:538]: `apt` is the more recent best practice! https://askubuntu.com/questions/445384/what-is-the-difference-between-apt-and-apt-get

**Comment by noah-weingarden** [comment]
Any reason you don't introduce students to Org Mode in this tutorial? I'm still loyal to the Cult of Vim, but my casual passes through the modern Emacs community suggest that Org Mode is one of its more beloved features.

**Comment by awdeorio** [comment]
> Any reason you don't introduce students to Org Mode in this tutorial?

The only reason is that I'm not personally proficient with Org Mode.  It's on my bucket list to play around it.

**Comment by noah-weingarden** [comment]
Ngl, you seem like the kind of person it would appeal to a lot!

**Pull request merged**